pydantic/main.py

```diff
@@ -339,17 +357,18 @@ def __init__(__pydantic_self__, **data: Any) -> None:
         Raises ValidationError if the input data cannot be parsed to form a valid model.
         """
         # Uses something other than `self` the first arg to allow "self" as a settable attribute
-        if TYPE_CHECKING:
-            __pydantic_self__.__dict__: Dict[str, Any] = {}
-            __pydantic_self__.__fields_set__: 'SetStr' = set()
         values, fields_set, validation_error = validate_model(__pydantic_self__.__class__, data)
         if validation_error:
             raise validation_error
-        object.__setattr__(__pydantic_self__, '__dict__', values)
-        object.__setattr__(__pydantic_self__, '__fields_set__', fields_set)
+        object_setattr(__pydantic_self__, '__dict__', values)
+        object_setattr(__pydantic_self__, '__fields_set__', fields_set)
+        __pydantic_self__._set_default_private_attributes(__pydantic_self__.__private_attributes__)
```

Comment:
why is this required?

Reply:
what do you mean exactly?

Reply:
Can we avoid this line since this is the critical path in terms of performance?

Reply:
It sets defaults of `__private_attributes__`, without this line `default` not `default_factoru` wouldn't work.
What's critical here in terms of performance? Function call itself or executing code of the function?

Reply:
Ok, I guess we need it.

Initialising/validating models is the critical path, I don't want to slow that down unless we absolutely have to.